src/D2L.Security.OAuth2/Keys/PrivateKeyData.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+
+namespace D2L.Security.OAuth2.Keys {
+	public sealed class PrivateKeyData {
+		internal static class KeyKinds {
+			public const string Rsa = "rsa";
+			public const string Ecdsa = "ecdsa";
+		}
+
+		public PrivateKeyData(

```

Comment:
This has to be `public` so that it can be deserialized by the implementer of `IPrivateKeyDataProvider`.